jscomp/ml/typecore.ml

```diff
@@ -591,6 +591,62 @@ let build_or_pat env loc lid =
           pat pats in
       (path, rp { r with pat_loc = loc },ty)
 
+let build_or_pat_for_variant_spread env loc lid expected_ty =
+  let path, decl = Typetexp.find_type env lid.loc lid.txt in
+  match decl with
+  | {type_kind = Type_variant constructors} -> (
+    (* TODO: Probably problematic that we don't account for type params here? *)
```

Comment:
This whole fn would be good to get your eyes on @cristianoc . It's mostly copied and adapted to work on regular variants from the existing `build_or_pat`, that operates on polyvariants.